modules/user_accounts/php/edit_user.class.inc

```diff
@@ -202,7 +202,11 @@ class Edit_User extends \NDB_Form
         $permIDs = array();
         // store the permission IDs
         if (!empty($values['permID'])) {
-            $permIDs = array_keys($values['permID']);
+            // An editor can only grant permission that he is granted of.
+            $permIDs = array_intersect(
+                array_keys($values['permID']),
+                $editor->getPermissionIDs()
+            );
```

Comment:
What happens if you don't have a permission and edit a user who does have the permission to try and grant the a different perm?

Reply:
It doesn't show up. It only shows the permissions you have been granted.

Reply:
But does it delete that permission (which should have been unaffected) from the user?

Reply:
@ZainVirani did you test this part? If not please remove Passed Manual Tests

Reply:
Oh I see, sorry I misunderstood. Will test now